src/components/Header/AddEditDialog/AddEditDialog.tsx

```diff
@@ -23,22 +28,16 @@ export const AddEditDialog = (props: AddEditDialogProps): JSX.Element => {
     return null;
   }
 
-  const handleInputChange = (event: any): void => {
-    const propName = event.target.id;
-    let propValue = event.target.value;
-    if (propName === 'genres') {
-      propValue = propValue.split(',');
-    }
-    dispatch(moviesActions.setMovie({ ...movieItem, [propName]: propValue }));
-  };
-
-  const onSubmitClick = (): void => {
-    if (props.isCreateMode) {
-      createMovie(movieItem);
-    } else {
-      updateMovie(movieItem);
-    }
-    props.handleCancelClick();
+  const onSubmitClick = (movie: MovieItemModel): void => {
+    dispatch(moviesActions.setMovie({ id: movieItem.id, ...movie }));
```

Comment:
1. Since the form state becomes local with formik, does it still make sense to dispatch movie update to the store?
2. Why createMovie/updateMovie are wrapped with a timeout? They do not seem to be dependent on movie state.
3. `0` parameter can be removed

Reply:
1. In my case yes, because we still use redux for api calls.
2. it won't work otherwise
3. sure